python-package/xgboost/spark/core.py

```diff
@@ -775,6 +771,12 @@ def _train_booster(pandas_df_iter):
 
             gpu_id = None
 
+            # If cuDF is not installed, then using DMatrix instead of QDM,
+            # because without cuDF, DMatrix performs better than QDM.
+            # Note: Checking `is_cudf_installed` in spark worker side because
+            # spark worker might has different python environment with driver side.
+            use_qdm = use_hist and is_cudf_installed()
```

Comment:
`use_hist` includes both cpu hist and gpu hist.

Reply:
Please fix the linter error as well.

Reply:
> use_hist includes both cpu hist and gpu hist.

Yes. Any issues on this ?

Reply:
CPU hist should continue to use qdm with or without cudf. cuDF is a GPU only library.

Reply:
ok got it.

Reply:
PR updated!